tools/shared.py

```diff
@@ -2508,20 +2506,24 @@ def closure_compiler(filename, pretty=True, advanced=True, extra_closure_args=[]
 
       # Web environment specific externs
       if Settings.target_environment_may_be('web') or Settings.target_environment_may_be('worker'):
-        BROWSER_EXTERNS_BASE = path_from_root('third_party', 'closure-compiler', 'browser-externs')
-        BROWSER_EXTERNS = os.listdir(BROWSER_EXTERNS_BASE)
-        BROWSER_EXTERNS = [os.path.join(BROWSER_EXTERNS_BASE, name) for name in BROWSER_EXTERNS
-                           if name.endswith('.js')]
-        CLOSURE_EXTERNS += BROWSER_EXTERNS
-
-      # Something like this (adjust memory as needed):
-      #   java -Xmx1024m -jar CLOSURE_COMPILER --compilation_level ADVANCED_OPTIMIZATIONS --variable_map_output_file src.cpp.o.js.vars --js src.cpp.o.js --js_output_file src.cpp.o.cc.js
+        try:
+          BROWSER_EXTERNS_BASE = path_from_root('third_party', 'closure-compiler', 'browser-externs')
+          BROWSER_EXTERNS = os.listdir(BROWSER_EXTERNS_BASE)
+          BROWSER_EXTERNS = [os.path.join(BROWSER_EXTERNS_BASE, name) for name in BROWSER_EXTERNS
+                             if name.endswith('.js')]
+          CLOSURE_EXTERNS += BROWSER_EXTERNS
+        except Exception:
+          # Currently we do not actually have any browser-externs files, so above will throw as
+          # git does not keep empty directories in repositories. But in future we are likely to add
+          # some, so keep this code around.
+          logger.debug('No browser-externs files found')
+          pass
```

Comment:
Can we just check for the existence of the directory rather than try/catch?:

```
BROWSER_EXTERNS_BASE = path_from_root('third_party', 'closure-compiler', 'browser-externs')
if os.path.exists(BROWSER_EXTERNS_BASE):
   BROWSER_EXTERNS = os.listdir(BROWSER_EXTERNS_BASE)
   ...
￼   CLOSURE_EXTERNS += BROWSER_EXTERNS
```

Reply:
Also no sure why all these variable are capitalized but I guess that's all existing code.